versioned/mongodb/src/main/java/com/dremio/nessie/versioned/impl/CodecProvider.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.dremio.nessie.versioned.impl;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.bson.BsonBinary;
+import org.bson.BsonReader;
+import org.bson.BsonWriter;
+import org.bson.codecs.Codec;
+import org.bson.codecs.DecoderContext;
+import org.bson.codecs.EncoderContext;
+import org.bson.codecs.configuration.CodecRegistry;
+
+import com.dremio.nessie.versioned.store.Id;
+import com.dremio.nessie.versioned.store.mongodb.codecs.EntityCodec;
+
+/**
+ * Codecs provide the heart of the SerDe process to/from BSON format.
+ * The Codecs are inserted into a CodecRegistry. However they may require the CodecRegistry to do their job.
+ * This apparent two way interdependency is resolved by using a CodecProvider.
+ * The CodecProvider is a factory for Codecs.
+ */
+@SuppressWarnings("unchecked")
```

Comment:
Why is this set at the class level? It seems like it shouldn't be needed at all. If needed, let's keep it as narrow as possible.